### PR TITLE
updateTelemetry on BaseSubsystem

### PR DESCRIPTION
**Pull request by aschokking**
* Add an optional `updateTelemetry` method on BaseSubsystem
* Add ability for these subsystems to be registered with BaseRobot class and they will be called automatically during every periodic loop.

The goal here was to make it as low overhead as possible, subsystems just need to override the method and then get registered (so no need for additional boilerplate commands to call updateTelemetry).

### CODE REVIEW

File: src/xbot/common/command/BaseRobot.java [hunk 1]
```diff
@@ -34,6 +36,8 @@
     protected Injector injector;
     
     protected Command autonomousCommand;
+    
+    protected Vector<BaseSubsystem> telemetryUpdatingSubsystems = new Vector<BaseSubsystem>();
```
Comment (WasabiFan) [inline src/xbot/common/command/BaseRobot.java]: Is there a reason that this is a `Vector`? They were deprecated in Java 2 IIRC. It seems like this should be a normal `ArrayList`.

File: src/xbot/common/command/BaseSubsystem.java [hunk 2]
```diff
@@ -13,5 +14,10 @@ public void setDefaultCommand(Command command) {
     protected void initDefaultCommand() {
 
     }
+    
+    public void updateTelemetry() {
```
Comment (WasabiFan) [inline src/xbot/common/command/BaseSubsystem.java]: This architecture feels odd. Not all subsystems have telemetry to report and some things that _aren't_ subsystems may want to supply telemetry, so putting it on the subsystem doesn't make much sense to me. Could we define a `TelemetrySource`, `ScheduledTelemetrySource` or similar interface that has this method on it and then implement that on the subsystems that want to be given telemetry cycles? Then the `BaseRobot` would operate on generic `TelemetrySource`s (which could be, for example, my vision server thread).

File: src/xbot/common/command/BaseRobot.java [hunk 3]
```diff
@@ -118,4 +122,19 @@ public void teleopPeriodic() {
     public void testPeriodic() {
         LiveWindow.run();
     }
+    
+    protected void sharedPeriodic(){
```
Comment (WasabiFan) [inline src/xbot/common/command/BaseRobot.java]: Space before brace

File: src/xbot/common/command/BaseRobot.java (hunk 1 above)
Comment (aschokking) [inline src/xbot/common/command/BaseRobot.java]: I'm just old :D

File: src/xbot/common/command/BaseSubsystem.java (hunk 2 above)
Comment (aschokking) [inline src/xbot/common/command/BaseSubsystem.java]: That's fair, I was debating between the two options and had leaned this way to reduce the number of things people had to learn about (like another interface they need to add to their subsystem). That said, the non-subsystem class case seems like a reasonable one.

File: src/xbot/common/command/BaseSubsystem.java [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import edu.wpi.first.wpilibj.command.Command;
 import edu.wpi.first.wpilibj.command.Subsystem;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```
Comment (WasabiFan) [inline src/xbot/common/command/BaseSubsystem.java]: ಠ_ಠ

**Pull request merged**